lib/rubocop/cop/corrector.rb

```diff
@@ -135,11 +141,21 @@ def remove_leading(range, size)
       # @param [Parser::Source::Range] range
       # @param [Integer] size
       def remove_trailing(range, size)
+        validate_range range
         to_remove = Parser::Source::Range.new(range.source_buffer,
                                               range.end_pos - size,
                                               range.end_pos)
         @source_rewriter.remove(to_remove)
       end
+
+      private
+
+      # :nodoc:
+      def validate_range(range)
+        return if range.source_buffer == @source_buffer
+        raise 'Correction range target buffer differs from ' \
```

Comment:
Can you include here the names for the two buffers? I think that's going to make debugging much easier.

Reply:
Ok, done. I doubt names will not do much, but additional check for buffer being a buffer at all - should.